From ce723673d8823adf2248771a7ea129622054a045 Mon Sep 17 00:00:00 2001
Message-Id: <ce723673d8823adf2248771a7ea129622054a045.1530775180.git.maciej.zenon.borzecki@canonical.com>
From: Maciej Borzecki <maciej.zenon.borzecki@canonical.com>
Date: Thu, 5 Jul 2018 07:28:41 +0200
Subject: [PATCH] dirs: improve distro detection for Antegros

Previous attempt to use ID_LIKE=archlinux for detecting Arch Linux like distros
failed. It seems that there is no consensus about using ID/ID_LIKE.

For reference, this is what is contained in /etc/os-release for Arch like
distros:

- Arch:
    ID=arch
    ID_LIKE=archlinux

- Manjaro (no ID_LIKE):
    ID=manjaro

- Antegros:
    ID="antergos"
    ID_LIKE="arch"

Relying on ID_LIKE seems to have little sense since there seems to be no
agreement as to what that variable should be set to.

Signed-off-by: Maciej Borzecki <maciej.zenon.borzecki@canonical.com>
---
 dirs/dirs.go | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dirs/dirs.go b/dirs/dirs.go
index 16a3a7a4f32c5e830254d160f395520505d6dd62..c37733ff250ab049aa18f60a54dd768dbe2b04d7 100644
--- a/dirs/dirs.go
+++ b/dirs/dirs.go
@@ -179,7 +179,7 @@ func SetRootDir(rootdir string) {
 	}
 	GlobalRootDir = rootdir
 
-	if release.DistroLike("fedora", "arch", "manjaro") {
+	if release.DistroLike("fedora", "arch", "manjaro", "antegros") {
 		SnapMountDir = filepath.Join(rootdir, "/var/lib/snapd/snap")
 	} else {
 		SnapMountDir = filepath.Join(rootdir, defaultSnapMountDir)
-- 
2.18.0

